ignite/src/jepsen/ignite/bank.clj

```diff
@@ -1,122 +1,146 @@
 (ns jepsen.ignite.bank
-  "Simulates transfers between bank accounts"
-  (:refer-clojure :exclude [test])
-    (:require [clojure.tools.logging   :refer :all]
-              [jepsen [ignite          :as ignite]
-                      [checker         :as checker]
-                      [client          :as client]
-                      [nemesis         :as nemesis]
-                      [generator       :as gen]]
-              [clojure.core.reducers :as r]
-              [jepsen.checker.timeline :as timeline]
-              [knossos.model           :as model]
-              [knossos.op :as op])
-    (:import client.Bank
-              (org.apache.ignite.transactions TransactionTimeoutException)
-              (org.apache.ignite.cache CacheMode CacheAtomicityMode CacheWriteSynchronizationMode)))
+    "Simulates transfers between bank accounts"
+    (:refer-clojure :exclude [test])
+    (:require [clojure.tools.logging :refer :all]
+      [jepsen [ignite :as ignite]
+       [checker :as checker]
+       [client :as client]
+       [nemesis :as nemesis]
+       [generator :as gen]]
+      [clojure.core.reducers :as r]
+      [jepsen.checker.timeline :as timeline]
+      [knossos.model :as model]
+      [knossos.op :as op])
+    (:import (org.apache.ignite Ignition)
+      (org.apache.ignite.transactions TransactionConcurrency TransactionIsolation)
+      (org.apache.ignite.transactions TransactionTimeoutException)
+      (org.apache.ignite.cache CacheMode CacheAtomicityMode CacheWriteSynchronizationMode)))
 
-(def accounts 10)
+(def n 10)
 (def account-balance 100)
 
+(def cache-name "ACCOUNTS")
+
+(defn read-values [cache n]
+      (vals (.getAll cache (set (range 0 n)))))
+
+(defn read-values-tr [ignite cache n transaction-concurrency transaction-isolation]
+      (with-open [tr (.txStart (.transactions ignite) transaction-concurrency transaction-isolation)]
+                 (let [values (read-values cache n)]
+                      (.commit tr)
+                      values)))
+
 (defn bank-checker
-  "Balances must all be non-negative and sum to the model's total."
-  []
-  (reify checker/Checker
-    (check [this test history opts]
-      (let [bad-reads (->> history
-                        (r/filter op/ok?)
-                        (r/filter #(= :read (:f %)))
-                        (r/map (fn [op]
-                          (let [balances (:value op)]
-                            (cond
-                              (not= accounts (count (keys balances)))
-                              {:type :wrong-n
-                               :expected accounts
-                               :found    (count (keys balances))
-                               :op       op}
+      "Balances must all be non-negative and sum to the model's total."
```

Comment:
By the way, you might be able to save the trouble of writing these checkers by hand by using jepsen.tests.bank. :-)